### DISCUSSION TITLE
Tab autocomplet for wordlist not working in zsh

### DISCUSSION THREAD

**Discussion by secure-77** [epi052/feroxbuster#405] [Q&A] [answered]
Hey,

Not sure if this is a bug of zsh-autosuggestions or feroxbuster or maybe i just miss some configuration.

I am using "oh my zsh", some time ago i installed the zsh-autosuggestions plugin. I thought everything works fine but then i noticed that the default tab auto completion with feroxbuster stopped working.

I am not 100% sure this is related to the zsh-autosuggestions plugin but i am sure before i activated this, the tab auto completion works fine. But when i deactivate this plugin and restart the shell the default tab auto completion still does not work.

Example, if i type
```
feroxbuster -w /usr/share/wor
```
and then TAB, i should be able to auto complete the path with  /usr/share/wordlist .. etc. but nothing happens.

in other tools like ffuf this works (also with zsh-autosuggestions)

I read in the readme, that on the installation of feroxbuster via APT there happens some stuff to provide autocompletion, maybe this is now lost? How can i check or repair this?

Site question: I am using feroxbuster 2.3.3, there should be the 2.4 but apt update & apt upgrade doesent provide me the version 2.4, is this because the push to the repos need more time and i just have to wait to get the latest version via APT?

Thanks for any hints or support :)

BR,
Sec77

**Comment by epi052** [comment]
Unfortunately, I don't use zsh, so don't really have a good strategy for debugging conflicts like that (if there is one).

The feroxbuster completions for zsh live at `/usr/share/zsh/vendor-completions/_feroxbuster` if you installed via apt. If that file isn't there, you can download it from [here](https://github.com/epi052/feroxbuster/blob/main/shell_completions/_feroxbuster) and put it in that same location. I don't know what that plugin does to existing completion scripts, but it may have altered it in some way.  If you get them both working together, lmk and we can get w/e mods necessary into the repo.

I'm not sure when you tried to update, but I got notified at 5am this morning that the kali repos were updated to include 2.4.0. The repos lag behind github releases by a few days, but even that turn around time is spectacular, imo.

**Comment by secure-77** [comment]
Thanks for your reply!

first answer to your questions:
- /usr/share/zsh/vendor-completions/_feroxbuster is available and the original from release 2.3.3
- there is still no update for feroxbuster via apt (maybe cause iam in germany and using other repo servers...) 


ok.. after lots of investigation i can break it down to compinit and the cache file(s)

As far as i understand compinit tries to parse and validate all parameters of a tool.
It seems that a `-` (dash) in any parameter will prevent the default auto completion and force to show all available parameter options of the tool if you press (double) TAB.

This is not feroxbuster related as it is valid for every other tool which is listed in any of the zcomdump files.

Simple sanity check with md5sum

in the default bash (there is no compinit), the tab autocomplete on path is working without any problems
```bash
md5sum -b /usr/share...
```
in zsh and oh my zsh, this is not working
but if i replace the dash with a slash
```bash
md5sum /b /usr/share...
```
is its working also in zsh

further more:
if i use the default .zshrc config and disable the line
```
compinit -d ~/.cache/zcomdump
```
the directory auto completion works again with `-` in parameters.

further 2:
i found 3 zcomdump files
~zcomdump
~zcomdump-hostname-5.8
~/.cache/zcomdump

if i delete the line 
```
'md5sum _md5sum'
```
in all of this dump files, the tab auto completion works again for md5sum. Of course, then the parameter suggestion is not working anymore. as feroxbuster is too in these dump files this is why it is also not working in feroxbuster

.. so far 

no it would be great if anybody know how i can configure compinit and oh my zsh that it behaves like the default bash auto completion. because the default bash can both:
complete directories even there is dash in the parameter before AND provide the parameters of the tool if you hit TAB after one dash 

but as this is not feroxbuster related, i think i have to ask in some zsh related section..

BR,
Sec77

**Comment by secure-77** [comment, marked as answer]
i have some news:

the thing about the md5sum tool was a recent issue 🙄 and is fixed in the next release. i took a look what they changed to fix the md5sum problem: https://github.com/zsh-users/zsh/commit/c843af635a470f209e4aca70b26fcd74ed31071b

playing a little bit with the `_feroxbuster` script i figured out that this little change will fix the problem for me:

changing this
```bash
   if is-at-least 5.2; then
        _arguments_options=(-s -S -C )
    else
        _arguments_options=(-s -C)
    fi
```
to this
```bash
   if is-at-least 5.2; then
        _arguments_options=(-s -S -C '*: :_files')
    else
        _arguments_options=(-s -C '*: :_files')
    fi
```

i can make a pull request if you want but i am not familiar with this script and syntax (it was more try and error to figure that out) so i dont know if this is ok or break something else  😅 

@epi052 just let me know
